### ISSUE TITLE
Dead GitHub Link on Rally Docs

### ISSUE THREAD

**Issue by hatdropper1977** [elastic/rally#386]
Hello - I went to the main Doc page for ESRALLY:
https://esrally.readthedocs.io/en/stable/index.html

I then click 'Edit On Github' in the top right corner, which takes me to...
https://github.com/elastic/rally/blob/5b3a507f6106ee496bb29a859fe9d15a88d85d74/docs/index.rst

I click that link and it's a dead link.

It should point to
https://github.com/elastic/rally/blob/master/docs/install.rst

**Comment by danielmitterdorfer** [comment]
Thanks for the report. I checked a few things:

* Up until version 0.7.2, readthedocs has included the name of the tag in the URL (e.g. `0.7.2`). After that, it has used a git commit hash.
* The git commit hashes do not match any of the git commit hashes in the Rally repo. To me this looks like an issue with Readthedocs.
* I checked [potentially related issues in the Readthedocs repo](https://github.com/rtfd/readthedocs.org/issues?utf8=%E2%9C%93&q=is%3Aissue+sha) but nothing stood out to me immediately.
* I did a build of the Rally docs on a branch on readthedocs.org and it seems the build SHA reported by readthedocs is now correct again (it matches the commit id of my push on the branch).

In summary: It seems that this is an issue with the readthedocs infrastructure which is already fixed. Unfortunately, rebuilding the docs does not seem to change the SHA. So I'll keep this ticket open as a reminder and will check that the SHA is correct after the next Rally release.

**Comment by danielmitterdorfer** [comment]
The problem persists after releasing Rally 0.9.0. Meanwhile I have analyzed the build logs from readthedocs and have raised https://github.com/rtfd/readthedocs.org/issues/3530 now.

**Comment by danielmitterdorfer** [comment]
The project team has pointed me to https://github.com/rtfd/readthedocs.org/issues/1820. The issue is long-standing and specifically related to the `stable` version link. The gist of the discussion is that the SHA in the link is the one of the git tag which is not recognised by Github.

Unfortunately, the configuration on Readthedocs overrides the respective user configuration so we can neither fix the link target nor disable the "Edit on Github" link. As a workaround I have now copied the original breadcrumbs file from their theme and modified the "Edit on Github" link to always point to master (this is actually better because we'd not backport doc changes anyway). In case the upstream issue gets fixed we can remove this workaround again. The `stable` link should be fixed with the next release of Rally.

**Comment by danielmitterdorfer** [comment]
The first build for `latest` (which corresponds) to `master` just succeeded and the results are at http://esrally.readthedocs.io/en/latest/. You won't notice any difference though because this version has always pointed at `master` but this is the final verification that the modified template is fine.
